packages/block/src/helpers.ts

```diff
@@ -7,7 +7,7 @@ import type { BlockHeaderBuffer, HeaderData } from './types'
  * @param {string} input string to check, convert, and return
  */
 export const numberToHex = function (input?: string) {
-  if (isFalsy(input)) return undefined
+  if (input === undefined) return undefined
```

Comment:
Yeah I think it's fine to make that method a bit more restrictive since that doesn't break anything. It would be a bit weird any to rely on the "empty string returns undefined" behavior of the pre-PR method